pyproject.toml

```
@@ -14,7 +14,7 @@ requires-python = ">=3.7"
 dependencies = [
     "anyconfig>=0.10.0",
     "attrs>=21.3",
-    "build>=0.7.0",
+    "build>=0.7.0,<1.0.0",
```

What about

```suggestion
    "build>=0.7.0,<1.0.0 ; python_version<'3.8'",
    "build>=0.7.0 ; python_version>='3.8'",
```

(didn't test this syntax, please double check)